docs/proposals/secrets-proposal.md

```diff
@@ -27,42 +27,42 @@ To create a new secret, a top level `secrets` key can be used as follows:
 version: "0.1-dev"
 
 services:
-   name: foobar
+ - name: foobar
    containers:
    - image: foo/bar:tag
      ports:
      - port: 8080:80
      env:
-     - foo=bar
-     - travis_creds:
-         secret: ci_secret/travis
+     - name: foo
+       value: bar
+     - name: travis_creds
+       secretRef: ci_secret/travis
      mounts:
      - mountPath: /etc/jenkins.conf
-       secret: ci_secret/jenkins
+       secretRef: ci_secret/jenkins
 
 secrets:
-    ci_secret:
-    - file: /tmp/travis.password
-    - data: jenkins=strongpassword@@
+  - name: ci_secret
+    fromFile:
+    - /tmp/travis.password
+    fromLiteral:
+    - jenkins=strongpassword@@
 ```
 
 The syntax will look like -
 
 ```yaml
 secrets:
-  secret_name:
-  - file: <file path>
-  - data: <data key>=<data value>
-  - data: <data key>=<base64 encoded data>
-    type: base64
-  - data: <data key>=<data value>
-    type: plaintext (default)
-
+  - name: <secret name>
+    fromFile:
+    - <file path>
+    - <file path>
+    fromLiteral:
```

Comment:
@kadel agree, I took these from from the `kubectl` command line flags for secrets, which are `--from-file` and `--from-literal`.

`data` is fine but it might confuse with the `secrets.data` field in Kubernetes secrets. But yes, we could go ahead with `data` since it's a generic enough term.

Also, if we go ahead with `data`, then I'd also want the `from` prefix to be removed from the file so the fields look the same a bit, i.e. `data` and `file` instead of `data` and `fromFile`.

WDYT?